data/src/main/java/io/gnosis/data/models/TransferDetails.kt

```diff
@@ -5,6 +5,8 @@ import io.gnosis.data.backend.dto.TxData
 import pm.gnosis.model.Solidity
 import java.util.*
 
+interface TransactionDetails
```

Comment:
Why do we not use a similar model structure as in the client gateway. A lot of information is duplicated here (which also shows in the view holder and mapping as a lot of code is duplicated).

The client gateway models were created with the UI design in mind. The top section which displays the transaction type specific details (transfer, settings or custom) is polymorphic and there you can put the specific parts for that into different layout. The other parts of the response are mapped to a section in the UI (status -> show in status section, if executed timestamp is present -> show in executed section, if txHash is present -> show Etherescan link, ...). The whole purpose of the client gateway is that the clients should be able to use nearly the same models as the backend returns and that in most cases no additional mapping is required. If this is not the case we should discuss the response data and check if it can be optimized.